NanoGPT_Engine.py

```diff
@@ -0,0 +1,201 @@
+"""
+Some example classes for people who want to create a homemade bot.
+
+With these classes, bot makers will not have to implement the UCI or XBoard interfaces themselves.
+"""
+
+from __future__ import annotations
+import re
+import chess
+from chess.engine import PlayResult
+import logging
+from homemade import ExampleEngine
+from lib.engine_wrapper import FillerEngine, check_for_draw_offer, get_book_move, get_egtb_move, get_online_move, move_time
+from nanogpt.nanogpt_module import NanoGptPlayer
+import chess.engine
+import chess.polyglot
+import chess.syzygy
+import chess.gaviota
+import logging
+import datetime
+import time
+import test_bot.lichess
+from lib import config, model, lichess
+from lib.timer import Timer, to_seconds
+from typing import Any, Optional, Union
+OPTIONS_TYPE = dict[str, Any]
+MOVE_INFO_TYPE = dict[str, Any]
+COMMANDS_TYPE = list[str]
+LICHESS_EGTB_MOVE = dict[str, Any]
+CHESSDB_EGTB_MOVE = dict[str, Any]
+MOVE = Union[chess.engine.PlayResult, list[chess.Move]]
+LICHESS_TYPE = Union[lichess.Lichess, test_bot.lichess.Lichess]
+
+from dataclasses import dataclass
+
+
+@dataclass
+class LegalMoveResponse:
+    move_san: Optional[str] = None
+    move_uci: Optional[chess.Move] = None
+    attempts: int = 0
+    is_resignation: bool = False
+    is_illegal_move: bool = False
+
+# Use this logger variable to print messages to the console or log files.
+# logger.info("message") will always print "message" to the console or log file.
+# logger.debug("message") will only print "message" if verbose logging is enabled.
+logger = logging.getLogger(__name__)
+
+
+# Return is (move_san, move_uci, attempts, is_resignation, is_illegal_move)
+def get_legal_move(
+    player: NanoGptPlayer,
+    board: chess.Board,
+    game_state: str,
+    player_one: bool = False,
+    max_attempts: int = 5,
+# ) -> LegalMoveResponse:
+) -> PlayResult:
+    """Request a move from the player and ensure it's legal."""
+    move_san = None
+    move_uci = None
+
+
+    for attempt in range(max_attempts):
+        move_san = player.get_move(game_state, 0.5)
```

Comment:
this temperature is hardcoded at 0.5. make this configurable. see below comment on passing in from argparser